src/app/(app)/product-list.tsx

```diff
@@ -14,65 +17,102 @@ import {
   View,
 } from '@/ui';
 
-const products: ProductCardProps[] = [
-  {
-    id: '1',
-    name: 'Baumler Company Wood Chair 1',
-    price: 36,
-    status: 'Restored',
-  },
-  {
-    id: '2',
-    name: 'Baumler Company Wood Chair 2',
-    price: 36,
-    status: 'Restored',
-  },
-  {
-    id: '3',
-    name: 'Baumler Company Wood Chair 3',
-    price: 36,
-    status: 'Restored',
-  },
-  {
-    id: '4',
-    name: 'Baumler Company Wood Chair',
-    price: 36,
-    status: 'Restored',
-  },
-  {
-    id: '5',
-    name: 'Baumler Company Wood Chair',
-    price: 36,
-    status: 'Restored',
-  },
-];
+type PageControllerProps = {
+  handlePrevPage: () => void;
+  handleNextPage: () => void;
+  pagination: Pagination;
+};
+
+const PageController = ({ props }: { props: PageControllerProps }) => {
+  return (
+    <View className="flex-row justify-between px-5 pb-2">
+      <TouchableOpacity
+        onPress={props.handlePrevPage}
+        disabled={props?.pagination.page === 1} // Disable previous button on first page
+        className={`rounded-3xl bg-dark_violet p-4 px-6`}
+      >
+        <Text className="text-lg text-white">Previous</Text>
+      </TouchableOpacity>
+      <TouchableOpacity
+        onPress={props.handleNextPage}
+        disabled={
+          props.pagination.next_url === '/api/v1/products?page=&items=7'
+        } // Disable next button if no next page
+        className={`rounded-3xl bg-dark_violet p-4 px-6`}
+      >
+        <Text className="text-lg text-white">Next</Text>
+      </TouchableOpacity>
+    </View>
+  );
+};
+
+const FiltersButton = () => {
+  return (
+    <View className="absolute bottom-28 flex w-auto items-center justify-center self-center pb-20">
+      <TouchableOpacity className="flex-row items-center gap-4 rounded-full bg-dark_violet p-4 px-8">
+        <Text className="text-lg font-bold text-white">Filers</Text>
+        <Image className="h-4 w-4" source={images.filterIcon()} />
+      </TouchableOpacity>
+    </View>
+  );
+};
+
+const SearchResult = () => {
+  return (
+    <View className="flex-row justify-between px-5 pt-6">
+      <Text className="text-lg text-gray-600">You searched for “chairs”</Text>
+      <TouchableOpacity>
+        <Text className="text-lg text-link">Clear all</Text>
+      </TouchableOpacity>
+    </View>
+  );
+};
 
 export default function ProducList() {
+  const [page, setPage] = useState(1);
+  const { data } = useProducts({ variables: { page: page } });
+  const [products, setProducts] = useState<Product[]>([]);
+
+  useEffect(() => {
+    if (data) {
+      setProducts(data.data);
+    }
+  }, [data]);
+
+  if (!data?.pagination) {
+    return;
+  }
+
+  const handleNextPage = () => {
+    if (data?.pagination.next_url) {
+      setPage((prevPage) => prevPage + 1);
+    }
+  };
+  const handlePrevPage = () => {
+    if (page > 1 && data?.pagination.prev_url) {
+      setPage((prevPage) => prevPage - 1);
+    }
+  };
```

Comment:
You could consider wrapping handleNextPage and handlePrevPage in useCallback to avoid unnecessary re-creation of the functions on each render.